### ISSUE TITLE
Hashcatify and Johnify fail if there is no salt.

### ISSUE THREAD

**Issue by hkopp** [anonion0/nsec3map#13]
If there is no salt in the NSEC3 hash, the commands fail.
Unfortunately i cannot provide you with example domains, due to customer NDAs.

```
$ ./hashcatify.py all.nsec3.zone.hash
...
Traceback (most recent call last):
  File "/home/user/gits_public/nsec3map/./hashcatify.py", line 6, in <module>
    n3map.hashcatify.main()
  File "/home/user/gits_public/nsec3map/n3map/hashcatify.py", line 41, in main
    sys.exit(hashcatify_main(sys.argv))
  File "/home/user/gits_public/nsec3map/n3map/hashcatify.py", line 32, in hashcatify_main
    salt = util.str_to_hex(nsec3.salt)
  File "/home/user/gits_public/nsec3map/n3map/util.py", line 18, in str_to_hex
    hex_list = ["{0:02x}".format(b) for b in struct.unpack('B'*len(s), s)]
TypeError: a bytes-like object is required, not 'str'
```

What made the error go away was the following change:

In the original hashcatify.py in line 32, we have
```
 32             salt = util.str_to_hex(nsec3.salt)
```

However, that fails if the nsec3.salt is empty. Therefore i changed it to:
```
            if nsec3.salt == '': 
                salt = ''
            else:
                salt = util.str_to_hex(nsec3.salt)
```

I do not know if this fixes the issue completely, i.e., if the hashes are valid, as i am no expert in the format of hashcat. Same with johnify.
If you would like me to submit a merge request, I can do so.

**Comment by anonion0** [comment]
Thanks.
Turns out to be another Python 3 migration oversight.
Fixed in 0b81c49.

With regards to john itself, it looks like it fails to recognize the hashes if there is no salt. I'll look into it.
I'm not sure about hashcat.

**Comment by anonion0** [comment]
[This PR](https://github.com/openwall/john/pull/5185) fixes the issue in john as well.
Thanks again for reporting this.
